### PR TITLE
[node] [client] Transfer error handling

### PR DESCRIPTION
**Pull request by LayneHaber**
- pings recipient to make sure theyre online
- rejects install on hub and client if there is insufficient collateral

### CODE REVIEW

File: modules/client/src/validation/appProposals.ts [hunk 1]
```diff
@@ -119,42 +119,46 @@ const baseAppValidation = async (
...
       app,
     )}`;
   }
 
-  if (!appWithNode && !app.intermediaries) {
-    return (
-      `Apps that are not with connected node should have ` +
-      `intermediaries. Proposed app: ${prettyLog(app)}`
-    );
+  if (isVirtual && !app.intermediaries) {
```
Comment (rhlsthrm) [inline modules/client/src/validation/appProposals.ts]: I think this needs to check `app.intermediaries.length > 0`, bc `[]` might be truthy.

File: modules/node/src/appRegistry/appRegistry.service.ts [hunk 2]
```diff
@@ -54,6 +58,38 @@ export class AppRegistryService implements OnModuleInit {
     );
   }
 
+  private async validateTransfer(params: NodeTypes.ProposeInstallParams): Promise<void> {
```
Comment (rhlsthrm) [inline modules/node/src/appRegistry/appRegistry.service.ts]: This logic is only valid if node is the intermediary right? Will it get the propose install event?

Comment (LayneHaber) [reply modules/node/src/appRegistry/appRegistry.service.ts]: Yeah you're right, it's only validating wrt to the recipient and recipients collateral, so any transfer app that would be installed as a ledger app would always fail. Fixed, and updated a lot of errors in the new changes

**Comment by rhlsthrm** [comment]
Tests are failing, we need to fix before merging.

**Comment by LayneHaber** [comment]
closing, reopening against staging

**Pull request closed without merging**